sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java

```diff
@@ -212,33 +206,67 @@ public SampleReferenceDto getReferenceByUuid(String uuid) {
-		Root<Sample> sample = cq.from(Sample.class);
-
-		Join<Sample, Sample> referredSample = sample.join(Sample.REFERRED_TO, JoinType.LEFT);
-		Join<Sample, Facility> lab = sample.join(Sample.LAB, JoinType.LEFT);
-		Join<Sample, Case> caze = sample.join(Sample.ASSOCIATED_CASE, JoinType.LEFT);
-		Join<Case, Person> cazePerson = caze.join(Case.PERSON, JoinType.LEFT);
-		Join<Case, Region> caseRegion = caze.join(Case.REGION, JoinType.LEFT);
-		Join<Case, District> caseDistrict = caze.join(Case.DISTRICT, JoinType.LEFT);
-
-		cq.multiselect(sample.get(Sample.UUID), 
-				caze.get(Case.EPID_NUMBER), sample.get(Sample.LAB_SAMPLE_ID), sample.get(Sample.SAMPLE_DATE_TIME), 
-				sample.get(Sample.SHIPPED), sample.get(Sample.SHIPMENT_DATE), sample.get(Sample.RECEIVED), sample.get(Sample.RECEIVED_DATE), 
-				sample.get(Sample.SAMPLE_MATERIAL), sample.get(Sample.SAMPLE_PURPOSE), sample.get(Sample.SPECIMEN_CONDITION), 
-				lab.get(Facility.UUID), lab.get(Facility.NAME), referredSample.get(Sample.UUID), 
-				caze.get(Case.UUID), cazePerson.get(Person.FIRST_NAME), cazePerson.get(Person.LAST_NAME),
-				caze.get(Case.DISEASE), caze.get(Case.DISEASE_DETAILS), 
-				caseRegion.get(Region.UUID), caseDistrict.get(District.UUID), caseDistrict.get(District.NAME), sample.get(Sample.PATHOGEN_TEST_RESULT), 
+		final CriteriaBuilder cb = em.getCriteriaBuilder();
+		final CriteriaQuery<SampleIndexDto> cq = cb.createQuery(SampleIndexDto.class);
+		final Root<Sample> sample = cq.from(Sample.class);
+
+		final QueryContext qc = new QueryContext(cb, cq, sample);
+
+		sampleService.buildJoins(qc, sampleCriteria);
+
+		final Join<Sample, Sample> referredSample      = qc.getJoin(Sample.class, Sample.class);
+
+		final Join<Sample, Facility> lab               = qc.getJoin(Sample.class, Facility.class);
+
+		final Join<Sample, Case> caze                  = qc.getJoin(Sample.class, Case.class);
+		final Join<Case, Person> casePerson            = qc.getJoin(Case.class, Person.class);
+		final Join<Case, Region> caseRegion            = qc.getJoin(Case.class, Region.class);
+		final Join<Case, District> caseDistrict        = qc.getJoin(Case.class, District.class);
+
+		final Join<Sample, Contact> contact            = qc.getJoin(Sample.class, Contact.class);
+		final Join<Contact, Person> contactPerson           = qc.getJoin(Contact.class, Person.class);
+		final Join<Contact, Region> contactRegion      = qc.getJoin(Contact.class, Region.class);
+		final Join<Contact, District> contactDistrict  = qc.getJoin(Contact.class, District.class);
+		final Join<Contact, Case> contactCase          = qc.getJoin(Contact.class, Case.class);
+		final Join<Case, Region> contactCaseRegion     = qc.getJoin(Case.class, Region.class, "contactCaseRegion");
+		final Join<Case, District> contactCaseDistrict = qc.getJoin(Case.class, District.class, "contactCaseDistrict");
+
+		final Expression diseaseSelect = qc.addExpression("disease",
```

Comment:
Never use hard-coded strings here because they might break if the entity fields are renamed; always use the String constants from the entity classes instead (e.g. Case.DISEASE)

Reply:
ok, actually here these are the aliases for the joins and expressions
but nonetheless they should not be hard coded extracted them to constants within the sample facade